test/unit/Serializer/ArrayDtoSerializerTest.php

```diff
@@ -0,0 +1,548 @@
+    {
+        return [
+            [
+                'payload' => [
+                    'queryParams' => [],
+                    'pathParams' => [],
+                    'bodyParams' => [],
+                ],
+                'expected' => [
+                    'queryParams' => ['firstParam' => 'SomeDefaultQueryParam'],
+                    'pathParams' => ['firstParam' => 'SomeDefaultPathParam'],
+                    'bodyParams' => [
+                        'firstParam' => 'SomeDefaultFirstBodyParam',
+                        'secondParam' => 0,
+                    ],
+                ],
+            ],
+            [
+                'payload' => [
+                    'queryParams' => [],
+                    'pathParams' => 'BadPathParams',
+                    'bodyParams' => [],
+                ],
+                'expected' => [
+                    'queryParams' => ['firstParam' => 'SomeDefaultQueryParam'],
+                    'pathParams' => ['firstParam' => 'SomeDefaultPathParam'],
+                    'bodyParams' => [
+                        'firstParam' => 'SomeDefaultFirstBodyParam',
+                        'secondParam' => 0,
+                    ],
+                ],
+            ],
+            [
+                'payload' => [
+                    'queryParams' => ['firstParam' => 'SomeCustomFirstQueryParam'],
+                    'pathParams' => ['firstParam' => 'SomeCustomFirstPathParam'],
+                    'bodyParams' => [
+                        'firstParam' => 'SomeCustomFirstBodyParam',
+                        'secondParam' => 1000,
+                    ],
+                ],
+                'expected' => [
+                    'queryParams' => ['firstParam' => 'SomeCustomFirstQueryParam'],
+                    'pathParams' => ['firstParam' => 'SomeCustomFirstPathParam'],
+                    'bodyParams' => [
+                        'firstParam' => 'SomeCustomFirstBodyParam',
+                        'secondParam' => 1000,
+                    ],
+                ],
+            ],
+            [
+                'payload' => [
+                    'queryParams' => [],
+                    'pathParams' => [],
+                    'bodyParams' => ['secondParam' => 1000],
+                ],
+                'expected' => [
+                    'queryParams' => ['firstParam' => 'SomeDefaultQueryParam'],
+                    'pathParams' => ['firstParam' => 'SomeDefaultPathParam'],
+                    'bodyParams' => [
+                        'firstParam' => 'SomeDefaultFirstBodyParam',
+                        'secondParam' => 1000,
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed[] $payload
+     * @param mixed[] $expected
+     *
+     * @throws Throwable
+     *
+     * @dataProvider createRequestDtoWithPathProvider
+     */
+    public function testCreateRequestDtoWithPath(array $payload, array $expected): void
```

Comment:
The exception thrown when the request contents is a resource should be tested too.
Take a look at the \Symfony\Component\HttpFoundation\Request::getContent method, there is an example of creating a resource from a string without reading files on the disk.

Reply:
Resource will never be returned here, since getContent will turn it into a string via stream_get_contents.

We can use mock here, but it seems overkill.

Added check for resource to provider.